Tests/test_file_tiff.py

```diff
@@ -57,11 +57,10 @@ def test_sanity(self, tmp_path):
 
     @pytest.mark.skipif(is_pypy(), reason="Requires CPython")
     def test_unclosed_file(self):
-        def open():
+        with pytest.warns(ResourceWarning):
             im = Image.open("Tests/images/multipage.tiff")
             im.load()
-
-        pytest.warns(ResourceWarning, open)
+            del im
```

Comment:
Why did you add this? Users should no longer delete images - https://pillow.readthedocs.io/en/stable/deprecations.html#image-del

Reply:
Because if the image is not `del`eted (or otherwise GC'd) without having been closed, the resource warning isn't triggered.

With the previous `lambda` syntax, the variable would've dropped out of scope and had the same effect.

Reply:
What about https://github.com/akx/Pillow/pull/4 instead?

Reply:
Fine by me. Merged that in here.